plugins/enthusiast-common/enthusiast_common/injectors/base.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from .document import BaseDocumentRetriever
+from .product import BaseProductRetriever
+
+
+class BaseInjector(ABC):
+    @property
```

Comment:
I'm wondering if that's a property, or do we want to make it explicit that every time you're getting a new object?